lib/OpenQA/Worker/Commands.pm

```diff
@@ -37,7 +37,7 @@ sub websocket_commands {
     }
     else {
         if (!$json->{type}) {
-            log_warning('Received WS message without type! ' . pp($json));
+            log_warning('Received WS message without type! ' . pp($json), channels => 'worker', default => 1);
```

Comment:
I have the feeling that this requires some more cleverness from the framework. It feels a bit weird that you have to channel every debug call in the worker . Why wouldn't log_warning know the default by itself?

And what is the purpose of default anyway?

Reply:
The *default* param is to specify to log also to the *default* logger.
When `OpenQA::Setup` object is created, **a default logger is created** according to the parameters passed and with the *worker.ini* configs

The developer might want to add more "channels" to log to some other places

This means that we have 3 possibilities for logging:

- log only to the default logger

- log only to some channel

- log to the default logger *and* some channel

So when someone do `log_debug("message")` it's going to log to the default log, which by default the log file inside the log folder configured in the workers.ini. **This is the default behaviour**

When someone do `log_debug("message", channels=>['chan1', 'chan2'])` it's going to log only to the loggers defined by both channels chan1 and chan2. **But not to the default log**

When someone do `log_debug("message", channels=>['chan1', 'chan2'], default => 1)` it's going to log only to the loggers defined by both channels chan1 and chan2 **and to the default log**

In the case you brought, we want the messages to be logged both to the worker-log.txt and to the default log.

Example:
 if you have no logging configuration in workers.ini and you launch the worker on the command line:
`perl /usr/share/openqa/script/worker --instance 1 --verbose`

- `log_debug("message")` 
This will write to the stdout only on the console that launched the worker

- `log_debug("message", channels=>['chan1', 'chan2'])`
This will write to wherever the channels chan1 and chan2 are pointing to, but nothing will appear in the console

- `log_debug("message", channels=>['chan1', 'chan2'], default => 1)` 
This will write to wherever the channels chan1 and chan2 are pointing to and to the console

Reply:
And do we really need that? To me --verbose is the switch to see all the messages generated from the worker on the console. 

Logging shouldn't be an art IMO

Reply:
`--verbose` only sets the default logger's log level to debug, the developer still needs to choose to where he wants to log.

But your previous comment made me think. Having to write `channels=>["chan1", "chan2"], default=>1` is a pain in the neck.

I'll implement a function where the developer can specify where to output by default, so doing 
`log_debug("message")` will be the equivalent of `log_debug('message', channel=>'asd')`  with the latter form used to overwrite the default settings for that call.

Is it better?

Reply:
I'm still wondering why in lib/OpenQA/Worker/Commands.pm we have to specify workers as channel. Where else would the worker log to?

Reply:
By default everything that a worker does is logged to 

`$ENV{WORKER_LOG_DIR}/machineName-instance.log`

https://github.com/os-autoinst/openQA/blob/master/lib/OpenQA/Setup.pm#L86

This allows to have a shared worker_log_dir with all the worker logs there. Since we also want that info to be uploaded, that is written to another file as well.

Reply:
I don't really buy the need to duplicate this. But fine - as long as a log_debug("I'm a log message, get me out of here!")  in worker code does the proper thing ...

Reply:
Maybe @foursixnine and @mudler could provide their opinion in this as well.

Reply:
Maybe it's time to combine all the logging code into a singleton class/object?
```
use OpenQA::Log;

OpenQA::Log->add_channel('autoinst', path => 'autoinst-log.txt', level => 'debug');
OpenQA::Log->debug('what a mess!');
```

Reply:
The idea provided by @kraih is almost what i had in mind for, but is more going for a state-of-art solution and i do think it's out of the purpose of this PR (let's keep things going simple, and do that in a separate moment).

To avoid a complete rewrite of the logging mechanism in this set of changes, i'd go for treating stdout/stderr as other channels as well, and during the initialization phase setting them up based on the detected configuration (logging to stdout or not) and context (os-autoinst, openQA) in such way there is no need to refactor the other parts of the codebase that are actually calling log_debug, log_info,  etc.